### ISSUE TITLE
Accessibility issue: Heading levels should only increase by one on /primer/view_components/security/code-scanning

### ISSUE THREAD

**Issue by lindseywild** [lindseywild/accessibility-scanner#111]
## What
An accessibility scan flagged the element `<h4 data-view-component="true" class="color-fg-default mb-2">                    Sign in to GitHub
</h4>` on https://github.com/primer/view_components/security/code-scanning?query=is%3Aclosed+branch%3Amain because heading levels should only increase by one. Learn more about why this was flagged by visiting https://dequeuniversity.com/rules/axe/4.11/heading-order?application=playwright.

## Screenshot

A screenshot was captured when this issue was detected.

![Screenshot](https://raw.githubusercontent.com/lindseywild/accessibility-scanner/gh-cache/.screenshots/99df31c6-ceb2-41ef-be07-40fbbc7a1793.png)

> Screenshot ID: `99df31c6-ceb2-41ef-be07-40fbbc7a1793`
> [View in gh-cache branch](https://github.com/lindseywild/accessibility-scanner/blob/gh-cache/.screenshots/99df31c6-ceb2-41ef-be07-40fbbc7a1793.png)

To fix this, ensure the order of headings is semantically correct.

Specifically:

Fix any of the following:
-   Heading order invalid

## Acceptance Criteria
- [ ] The specific axe violation reported in this issue is no longer reproducible.
- [ ] The fix MUST meet WCAG 2.1 guidelines OR the accessibility standards specified by the repository or organization.
- [ ] A test SHOULD be added to ensure this specific axe violation does not regress.
- [ ] This PR MUST NOT introduce any new accessibility issues or regressions.
